plugins/modules/identity/bitwarden.py

```diff
@@ -0,0 +1,699 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2022, Jonathan Lung (@lungj) <lungj@heresjono.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+module: bitwarden
+author:
+    - Jonathan Lung (@lungj)
+requirements:
+    - C(bw) Bitwarden command line utility. See U(https://bitwarden.com/help/cli/)
+    - Organization names in Bitwarden are unique.
+    - Folders in Bitwarden have no duplicate names per organization.
+    - The name of items in folders uniquely identifies the items per organization.
+notes:
+    - Tested with C(bw) version 2022.6.2
+short_description: Manipulate items in Bitwarden
+description:
+    - M(community.general.bitwarden) wraps the C(bw) command line utility to modify Bitwarden-compatible vaults.
+    - A fatal error occurs if any of the items being searched for can not be found.
+    - Recommend using with the C(no_log) option to avoid logging the values of the secrets being retrieved.
+options:
+    cli_path:
+        type: path
+        description: Used to specify the exact path to the C(bw) command line interface
+        required: false
+        default: 'bw'
+    organization_name:
+        type: str
+        description:
+            - Name of organization to target.
+            - Leave as null for a wildcard vault or C('') for personal vaults.
+            - Mutually exclusive from organization_id.
+        required: false
+        default: null
+    organization_id:
+        type: str
+        description:
+            - UID of organization to target.
+            - Leave as null for a wildcard vault.
+            - Mutually exclusive from organization_name.
+        required: false
+        default: null
+    folder_name:
+        type: str
+        description:
+            - Name of folder to target.
+            - Leave as null for a wildcard folder.
+            - Use C('') or C('No Folder') for no folder.
+            - Mutually exclusive from folder_id.
+        required: false
+        default: null
+    folder_id:
+        type: str
+        description:
+            - UID of folder to target.
+            - Leave as null for a wildcard folder.
+            - Mutually exclusive from folder_name.
+        required: false
+        default: null
+    item_name:
+        type: str
+        description:
+            - Name of item to target.
+            - Mutually exclusive from item_id.
+        required: false
+        default: null
+    item_id:
+        type: str
+        description:
+            - UID of item to target.
+            - Mutually exclusive from item_name.
+        required: false
+        default: null
+    login:
+        type: dict
+        description:
+            - Mapping of keys to values for bw login details such as C(username) and C(password).
```

Comment:
Shouldn't the structure of this be predefined?

Reply:
I'm new to both Ansible module-writing and Bitwarden (I started both on Tuesday!). The documentation for the `bw` client suggests that users are supposed to fill out elements of the template returned by `bw get template item`; the structure does not _appear_ to be documented specifically. To provide maximum flexibility to maintain forwards/backwards compatibility, I decided not to specify any limits on fields here.

Reply:
@lungj many modules will have params like `login`, `password`, `token`. That seems to cover the majority of the cases (where authentication is needed). IMO going "free-form" in the name of a wished-future-backwards compatibility makes it more complicated than it needs to be. I mean, if that was to become a trend, we would never again have user and password parameters anywhere.

Just my 2 cents.